xarray_schema/dataarray.py

```diff
@@ -0,0 +1,206 @@
+        else:
+            self._dtype = DTypeSchema(value)
+
+    @property
+    def dims(self) -> Union[DimsSchema, None]:
+        return self._dims
+
+    @dims.setter
+    def dims(self, value):
+        if value is None or isinstance(value, DimsSchema):
+            self._dims = value
+        else:
+            self._dims = DimsSchema(value)
+
+    @property
+    def shape(self) -> ShapeSchema:
+        return self._shape
+
+    @shape.setter
+    def shape(self, value: Union[ShapeSchema, ShapeT, None]):
+        if value is None or isinstance(value, ShapeSchema):
+            self._shape = value
+        else:
+            self._shape = ShapeSchema(value)
+
+    @property
+    def chunks(self) -> ChunksSchema:
+        return self._chunks
+
+    @chunks.setter
+    def chunks(self, value):
+        if value is None or isinstance(value, ChunksSchema):
+            self._chunks = value
+        else:
+            self._chunks = ChunksSchema(value)
+
+    @property
+    def name(self) -> NameSchema:
+        return self._name
+
+    @name.setter
+    def name(self, value):
+        if value is None or isinstance(value, NameSchema):
+            self._name = value
+        else:
+            self._name = NameSchema(value)
+
+    @property
+    def array_type(self) -> ArrayTypeSchema:
+        return self._array_type
+
+    @array_type.setter
+    def array_type(self, value):
+        if value is None or isinstance(value, ArrayTypeSchema):
+            self._array_type = value
+        else:
+            self._array_type = ArrayTypeSchema(value)
+
+    @property
+    def checks(self) -> List[Callable]:
+        return self._checks
+
+    @checks.setter
+    def checks(self, value):
+        if value is not None:
+            if not all([callable(f) for f in value]):
+                raise ValueError('All checks must be callables')
+            self._checks = value
+        else:
+            self._checks = []
+
+    def validate(self, da: xr.DataArray) -> None:
+        '''Check if the DataArray complies with the Schema.
+
+        Parameters
+        ----------
+        da : xr.DataArray
+            DataArray to be validated
+
+        Returns
+        -------
+        xr.DataArray
+            Validated DataArray
+
+        Raises
+        ------
+        SchemaError
+        '''
+        if not isinstance(da, xr.DataArray):
+            raise ValueError('Input must be a xarray.DataArray')
+
+        if self.dtype is not None:
+            self.dtype.validate(da.dtype)
+
+        if self.name is not None:
+            self.name.validate(da.name)
+
+        if self.dims is not None:
+            self.dims.validate(da.dims)
+
+        if self.shape is not None:
+            self.shape.validate(da.shape)
+
+        if self.coords is not None:
+            raise NotImplementedError('coords schema not implemented yet')
+
+        if self.chunks is not None:
+            self.chunks.validate(da.chunks, da.dims, da.shape)
```

Comment:
why don't these get checked in the test? perhaps i'm not understanding how this testing works.

Reply:
These are not currently checked because we're testing the component schemas directly, not via the DataArraySchema class. But, testing through this route is easy so I'll add that too.